prov/usnic/src/usdf_endpoint.c

```diff
@@ -76,3 +77,54 @@ usdf_endpoint_open(struct fid_domain *domain, struct fi_info *info,
 		return -FI_ENODEV;
 	}
 }
+
+int usdf_ep_getopt_connected(fid_t fid, int level, int optname, void *optval,
+		size_t *optlen)
+{
+	size_t *cm_size;
+	size_t dest_size;
+
+	USDF_TRACE_SYS(EP_CTRL, "\n");
+
+	if (!optval || !optlen)
+		return -FI_EINVAL;
+
+	if (level != FI_OPT_ENDPOINT)
+		return -FI_ENOPROTOOPT;
+
+	switch (optname) {
+	case FI_OPT_CM_DATA_SIZE:
+		dest_size = *optlen;
```

Comment:
> Since we're being friendly to our users here in other ways, why not also include a NULL check before pointer deref?

Other than the one on lines 89-90?